enterprise/views.py

```diff
@@ -416,3 +419,121 @@ def post(self, request):
             # We're handing consent only for a particular course
             return self.post_course_specific_consent(request, specific_course, consent_provided)
         return self.post_account_consent(request, consent_provided)
+
+
+class LandingPageView(View):
+    """
+    Enterprise landing page view.
+
+    This view will display the course mode selection with related enterprise
+    information.
+    """
+
+    enterprise_welcome_text = _(
+        "{strong_start}{enterprise_customer_name}{strong_end} has partnered with "
+        "{strong_start}{platform_name}{strong_end} to offer you high-quality learning "
+        "opportunities from the world's best universities."
+    )
+    confirmation_text = _("Confirm your course")
+    starts_at_text = _("Starts")
+    course_pace_text = _("Paced")
+    view_course_details_text = _("View Course Details")
+    select_mode_text = _("Please select one:")
+    enroll_text = _("Enroll")
+    audit_text = _("Audit")
+    price_text = _("Price")
+    free_price_text = _("FREE")
+    discount_text = _("Discount provided by {enterprise_customer_name}")
+    not_eligible_for_cert_text = _(
+        "Not eligible for a certificate; does not count toward a MicroMasters"
+    )
+    continue_link_text = _("Continue")
+
+    def get_enterprise_landing_page(self, request, course_id):
+        """
+        Render enterprise specific course track selection page.
+
+        A 404 will be raised if any of the following conditions are met:
+            * Course details are not found against the provided course id.
+            * No enterprise customer uuid querystring "ec_uuid" in the request.
+            * No enterprise customer against the enterprise customer uuid in
+                the request querystring.
+
+        """
+        try:
+            client = CourseApiClient()
+            course_details = client.get_course_details(course_id)
+        except HttpClientError:
+            raise Http404
+
+        if course_details is None:
+            raise Http404
+
+        enterprise_customer_uuid = request.GET.get('ec_uuid')
+        if not enterprise_customer_uuid:
+            raise Http404
+
+        try:
+            enterprise_customer_uuid = UUID(enterprise_customer_uuid)
+            # pylint: disable=no-member
+            enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
+        except (ValueError, EnterpriseCustomer.DoesNotExist):
+            raise Http404
+
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
```

Comment:
since, we are using these variables only once, why not skip variable definitions and pass context data directly.
e.g.
```
context_data = {
     'page_title': _('Choose Your Track'),
     'page_language': get_language_from_request(request),
     'platform_name': configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME),
     'course_id': course_id,
    'course_name': course_details['name'],
    'course_organization': course_details['org'],
     ...
```
I think this will simply it.